Lesson 7

KayumovELdar opened the pull request.

EldarKa committed:
seed

EldarKa committed:
test_controller

EldarKa committed:
test_controller

EldarKa committed:
errors

EldarKa committed:
time

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -10,17 +10,31 @@ def index
+
+  def updale
+    @question = @test.questions.update(question_params)
+      if @question.save
```
ты уже сохраняешь вопрос в 30 строке. Для проверки используй update

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -10,17 +10,31 @@ def index
+      else
+        render :edit
+      end
+  end
```
не уверен что данный экшен работает. От куда тут может взяться тест? И какой вопрос пытаешься обновить?)

KayumovELdar replied:
как понимаю нужно подправить данный edit ?

VidgarVii commented on app/views/questions/_form.html.erb:
```diff
@@ -0,0 +1,14 @@
+    <%= form.text_field :text %>
+    <%= @question.errors.full_messages_for(:text).each do |message| %>
+      <div><%= message %></div>
+    <% end %>
```
а зачем тогда 2 строка?

VidgarVii commented on app/views/shared/_footer.html.erb:
```diff
@@ -0,0 +1,6 @@
+<script type="text/javascript">
+    document.write(Date());
+</script>
+<th><%= link_to "Test_guru",'https://github.com/KayumovELdar/test-guru-r', rel: :nofollow, target: :_blank %></th>
```
1 тут должен вызываться хелпер
2 `<th>` тут не уместен

VidgarVii replied:
**Для формирования ссылки на GitHub репозиторий: github_url(author, repo)**

VidgarVii commented on app/views/tests/show.html.erb:
```diff
@@ -0,0 +1,20 @@
+  <tr>
+    <th><%=question.text%></th>
+  </tr>
+<% end %>
```
1 используй render collection
2 нужны ссылки чтобы можно было совершать действия над вопросом.

EldarKa committed:
question update

EldarKa committed:
fix _form

EldarKa committed:
github_url

EldarKa committed:
th

EldarKa committed:
updale

EldarKa committed:
fix

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -10,17 +10,31 @@ def index
   end
 
+  def edit
+  end
```
сюда как попадает вопрос?)

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -10,17 +10,31 @@ def index
+  end
+
+  def updale
+    @question = Questions.find(params[:id])
```
у тебя ведь есть уже колбек для поиска вопроса

VidgarVii commented on app/controllers/tests_controller.rb:
```diff
@@ -6,4 +6,48 @@ def index
+  private
+
+    def find_test
+      @test = Test.find(params[:id])
```
по гугли DRY и shotgun operation )

VidgarVii commented on app/views/tests/index.html.erb:
```diff
@@ -0,0 +1,15 @@
+    <th>Name</th>
+    <th>Level</th>
+  </tr>
+  <% @tests.each do |test| %>
```
используй render collection

VidgarVii commented on app/views/shared/_footer.html.erb:
```diff
@@ -0,0 +1,6 @@
+<p>Eldar Kayumov</p>
+<script type="text/javascript">
+    document.write(Date());
+</script>
```
нужен helper на руби

VidgarVii commented on app/views/shared/_footer.html.erb:
```diff
@@ -0,0 +1,6 @@
+<script type="text/javascript">
+    document.write(Date());
+</script>
+<th><%=github_url('KayumovELdar', 'test-guru-r'%></th>
```
а куда сам метод пропал?

EldarKa committed:
fix

1.правка test_controller
2.ghfdrf question_controller
3.tests index (render tests)
4.shared _footer
4.1  добавлен current_year

EldarKa committed:
fix

EldarKa committed:
mini fix

EldarKa committed:
url

EldarKa committed:
find_test

VidgarVii commented on app/controllers/application_controller.rb:
```diff
@@ -1,2 +1,6 @@
+  private
+    def find_test
+      @test = Test.find(params[:test_id])
+    end
```
лучше объявляй метод там где действительно он нужен

KayumovELdar replied:
то есть в вопросе и тесте ???? но это же повторение)))))))))

KayumovELdar replied:
такое уточнение, дальше у нас примерно будет три блока схожих по содержанию. ответ, вопрос, и сам тест. Схоже будет создать удалить обновить итд.... я думаю сделать общий модуль с обобщенным текстом, только вопросс куда лучше будет данный файл поместить и через require_relative пришить

VidgarVii commented on app/controllers/tests_controller.rb:
```diff
@@ -1,9 +1,54 @@
+
+    def rescue_with_test_not_found
+    render plain: 'Test was not found!'
+    end
```
отступы съехали

VidgarVii commented on app/views/questions/show.html.erb:
```diff
@@ -1,4 +1,4 @@
-% question = Question.find(params[:id]) %>
+<% question = Question.find(params[:id]) %>
```
нужный вопрос должен приходить из контроллера

EldarKa committed:
fix

EldarKa committed:
rubocop

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -1,29 +1,43 @@
   def index
-      render html: @test.questions
+    render html: @test.questions
   end
```
**Список вопросов теста должен выводиться на той же странице, что и сам тест**

этот экшен не нужен

VidgarVii commented on app/controllers/tests_controller.rb:
```diff
@@ -1,9 +1,56 @@
+require_ralative 'Questions_controller'
+
 class TestsController < ApplicationController
+  require_ralative 'Questions_controller'
```
не надо ни каких require_ralative в rails

VidgarVii commented on app/views/questions/_form.html.erb:
```diff
@@ -0,0 +1,11 @@
+  <div>
+    <%= form.submit %>
+  </div>
+<% end %>
```
у тебя эта форма вызывается при обновлении вопроса. А для этого не нужен тест, соответственно и ссылка может быть некорректной
Пусть шапка формы создаётся отдельно для каждого макета. подлюкчай через партиал только поля формы

VidgarVii commented on app/views/shared/_errors.html.erb:
```diff
@@ -0,0 +1,10 @@
+<%= if resource.errors.present? %>
```
= не нужна

EldarKa committed:
fix

EldarKa committed:
fix

EldarKa committed:
fix

EldarKa committed:
Update _form.html.erb

EldarKa committed:
Update questions_helper.rb

EldarKa committed:
form

EldarKa committed:
Update questions_controller.rb

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -32,14 +42,14 @@ def question_params
   end
 
+  def find_question
+    @question = @test.questions.find(params[:id])
```
ряд экшенов не требует теста. Значит тут будет ошибка

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -1,29 +1,39 @@
   end
 
   def destroy
     @questioт.destroy
```
questioт

VidgarVii commented on app/controllers/questions_controller.rb:
```diff
@@ -1,29 +1,39 @@
   def destroy
     @questioт.destroy
-    render html: "Вопрос удалён!"
+    render html: 'Вопрос удалён!'
```
по ТЗ нужно реализовать представления (и соответсвенно редиректы и рендеринги) для всех экшенов
посмотри еще раз скринкаст
подобный render html не используется

EldarKa committed:
global fix

EldarKa committed:
rubocop

EldarKa committed:
mini Fix

EldarKa committed:
Update README.md

EldarKa committed:
Lesson_8

EldarKa committed:
Update application.rb
